code/game/objects/structures/tables_racks.dm

```diff
@@ -227,71 +227,91 @@
 /obj/structure/table/item_interaction_secondary(mob/living/user, obj/item/tool, list/modifiers)
 	if(istype(tool, /obj/item/construction/rcd))
 		return NONE
-
 	if(istype(tool, /obj/item/toy/cards/deck))
-		var/obj/item/toy/cards/deck/dealer_deck = tool
-		if(HAS_TRAIT(dealer_deck, TRAIT_WIELDED)) // deal a card faceup on the table
-			var/obj/item/toy/singlecard/card = dealer_deck.draw(user)
-			if(card)
-				card.Flip()
-				attackby(card, user, list2params(modifiers))
-			return ITEM_INTERACT_SUCCESS
+		. = deck_act(user, tool, modifiers, TRUE)
+
+	// Continue to placing if we don't do anything else
+	if(.)
+		return .
 
-	return item_interaction(user, tool, modifiers)
+	if(!user.combat_mode)
+		return table_place_act(user, tool, modifiers)
+
+	return NONE
 
 /obj/structure/table/item_interaction(mob/living/user, obj/item/tool, list/modifiers)
 	if(istype(tool, /obj/item/storage/bag/tray))
-		var/obj/item/storage/bag/tray/tray = tool
-		if(tray.contents.len > 0) // If the tray isn't empty
-			for(var/obj/item/thing in tray.contents)
-				AfterPutItemOnTable(thing, user)
-			tool.atom_storage.remove_all(drop_location())
-			user.visible_message(span_notice("[user] empties [tool] on [src]."))
-			return ITEM_INTERACT_SUCCESS
-		// If the tray IS empty, continue on (tray will be placed on the table like other items)
+		. = tray_act(user, tool)
+	else if(istype(tool, /obj/item/toy/cards/deck))
+		. = deck_act(user, tool, modifiers, FALSE)
+	else if(istype(tool, /obj/item/riding_offhand))
+		. = riding_offhand_act(user, tool)
 
-	if(istype(tool, /obj/item/toy/cards/deck))
-		var/obj/item/toy/cards/deck/dealer_deck = tool
-		if(HAS_TRAIT(dealer_deck, TRAIT_WIELDED)) // deal a card facedown on the table
-			var/obj/item/toy/singlecard/card = dealer_deck.draw(user)
-			if(card)
-				attackby(card, user, list2params(modifiers))
-			return ITEM_INTERACT_SUCCESS
-
-	if(istype(tool, /obj/item/riding_offhand))
-		var/obj/item/riding_offhand/riding_item = tool
-		var/mob/living/carried_mob = riding_item.rider
-		if(carried_mob == user) //Piggyback user.
-			return NONE
-		if(user.combat_mode)
-			user.unbuckle_mob(carried_mob)
-			tablelimbsmash(user, carried_mob)
-		else
-			var/tableplace_delay = 3.5 SECONDS
-			var/skills_space = ""
-			if(HAS_TRAIT(user, TRAIT_QUICKER_CARRY))
-				tableplace_delay = 2 SECONDS
-				skills_space = " expertly"
-			else if(HAS_TRAIT(user, TRAIT_QUICK_CARRY))
-				tableplace_delay = 2.75 SECONDS
-				skills_space = " quickly"
-			carried_mob.visible_message(span_notice("[user] begins to[skills_space] place [carried_mob] onto [src]..."),
-				span_userdanger("[user] begins to[skills_space] place [carried_mob] onto [src]..."))
-			if(do_after(user, tableplace_delay, target = carried_mob))
-				user.unbuckle_mob(carried_mob)
-				tableplace(user, carried_mob)
-		return ITEM_INTERACT_SUCCESS
+	// Continue to placing if we don't do anything else
+	if(.)
+		return .
```

Comment:
```suggestion
	if(. != NONE)
		return .
```

again, just don't wanna treat it like a boolean

Reply:
So I've been treating it like a boolean almost entirely for consistency, because our base item interaction code treats it in much the same way:
https://github.com/tgstation/tgstation/blob/d3f6c152839ec16f3ad2efd41ab20eed0fd8c457/code/game/atom/atom_tool_acts.dm#L41-L45
I'm fine doing it if you still think so, it does make sense, but that's why